src/autopas/utils/SimilarityFunctions.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file SimilarityFunctions.h
+ * @author J. Kroll
+ * @date 01.06.2020
+ */
+
+#pragma once
+
+#include "ThreeDimensionalMapping.h"
+#include "autopas/containers/ParticleContainerInterface.h"
+#include "autopas/molecularDynamics/MoleculeLJ.h"
+
+namespace autopas::utils {
+
+/**
+ * calculates homogeneity and max density of given AutoPas simulation
+ *
+ * @tparam Particle
+ * @param autopas
+ * @return  [homogeneity, max_density]
+ */
+template<class Particle>
+double static *calculateHomogeneityAndMaxDensity(const std::shared_ptr<autopas::ParticleContainerInterface<Particle>>& container) {
```

Comment:
same problem as above:
you are allocating memory on the heap but never free it.
Even more: you don't even need heap memory, so instead, just use two `double`s and return a `std::pair<double, double>` or, alternatively a `std::array<double ,2>`.

```suggestion
std::pair<double,double> calculateHomogeneityAndMaxDensity(const std::shared_ptr<autopas::ParticleContainerInterface<Particle>>& container) {
```

Reply:
solved with commit "use pair instead of pointer"